### ISSUE TITLE
Support OPTIONAL ARGUMENT in user defined macro by \newcommand

### ISSUE THREAD

**Issue by zq1997** [KaTeX/KaTeX#2228]
**Is your feature request related to a problem? Please describe.**
It seems that optional arguments are not supportes by user defined `\newcommand`.
Just see the following code.
```tex
\newcommand \foo [2] [bar] {{#1}^{#2}}
\foo {1}
\foo [baz]{2}
```
![image](https://user-images.githubusercontent.com/17429111/77845001-bc5acd80-71dd-11ea-9848-d85dec34f79f.png)

**Describe the solution you'd like:**
Would this optional argumant feature be implemented just like `\sqrt`macro? this is a predefined macro in `KaTeX` with optional argument support.

**Link to or name of a (La)TeX package that provides the same feature:**
N/A

**Describe alternatives you've considered:**
N/A

**Additional context:**
N/A

**Comment by edemaine** [comment]
Indeed, this shouldn't be too hard to implement, because KaTeX already internally supports optional arguments (without default values).  A small extension to `\newcommand` ought to do it.  A pull request would be welcome; otherwise, I can try to get to it eventually...

**Comment by zq1997** [comment]
I am not very familiar with this project, hope for the pull request in the future.

**Comment by klimanek** [comment]
Any update on this? Does KaTeX support optional argument now?

**Comment by edemaine** [comment]
Alas, no. When there are updates, they will appear here.

**Comment by aleeper** [comment]
Upvote, for what it's worth 😅
